Week1/SantiagoMedina/unit-test/src/test/java/com/globant/bootcamp/spring/unittest/kata/AccountServiceImplTest.java

```diff
@@ -0,0 +1,86 @@
+package com.globant.bootcamp.spring.unittest.kata;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import springbootcamp.unittest.kata.business.AccountServiceImpl;
+import springbootcamp.unittest.kata.data.AccountRepository;
+import springbootcamp.unittest.kata.domain.Account;
+import springbootcamp.unittest.kata.domain.AccountType;
+import java.lang.reflect.Constructor;
+import static org.mockito.Mockito.when;
+
+public class AccountServiceImplTest {
+
+    @Mock
+    private AccountRepository accountRepository;
+
+    @InjectMocks
+    private AccountServiceImpl accountService;
+
+    private Account currentAccount;
+    private Account savingAccount;
+    
+
+    @Before
+    public void setUp(){
+        MockitoAnnotations.initMocks(this);
+
+        try{
+            Constructor<Account> buildAccount = Account.class.getDeclaredConstructor(String.class, int.class, AccountType.class);
+            buildAccount.setAccessible(true);
+            currentAccount = buildAccount.newInstance("Santiago",0,AccountType.CURRENT);
+            savingAccount = buildAccount.newInstance("Felipe",0,AccountType.SAVING);
+
+        }
+        catch (Exception e){
+            System.out.println(e.getMessage());
+        }
+
+        when(accountRepository.findAccountByUsernameAndType("Santiago", AccountType.CURRENT)).thenReturn(currentAccount);
+        when(accountRepository.updateAccount(currentAccount)).thenReturn(currentAccount);
+
+        when(accountRepository.findAccountByUsernameAndType("Felipe", AccountType.SAVING)).thenReturn(savingAccount);
+        when(accountRepository.updateAccount(savingAccount)).thenReturn(savingAccount);
+    }
+
+    @Test
+    public void transferMoneyCurrentToSaving() {
+        
+        int balanceCurrentAccount = 3000;
+        int balanceSavingAccount = 1500;
+        int transferAmount = 2500;
+
+        currentAccount.setBalance(balanceCurrentAccount);
+        savingAccount.setBalance(balanceSavingAccount);
+
+        try {
+            accountService.transferMoney(currentAccount,savingAccount,transferAmount);
+        } catch (Exception e) {
```

Comment:
Same here, at least, your unit test is related to validate if an exception is thrown based on some condition.

Reply:
In this case is neccesary because when the there is insufficient funds the method "transferMoney" throw a Exception.